my_enumerable.rb

```diff
@@ -0,0 +1,124 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    idx = to_a
+    idx.size.times do |index|
+      yield idx[index]
+    end
+
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    idx = to_a
+    idx.size.times do |index|
+      yield idx[index], index
+    end
+
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    select = []
+    my_each do |i|
+      select << i if yield(i)
+      select
+    end
+  end
+
+  def my_all(arg = nil)
+    if block_given?
+      my_each.to_a {|idx| return false if yield(idx) == false}
+      return true
+    elsif arg.nil?
+      my_each.to_a do |idx| return false if idx = false or if idx.nil?
+
+    elsif !arg.nil && (arg.is_a? class
+      my_each.to_a {|idx| return false unless [item.class, item.class.superclass].include?(arg)}
+    elsif !arg.nil && arg.class == regexp
+      my_each.to_a {|idx| return false unless arg.match(idx)}
+    else
+      my_each.to_a {|idx| return false if idx != arg}
+    end
+    true
+  end
+
+  def my_any(arg = nil)
+    if block_given?
+      my_each.to_a {|idx| return true if yield(idx)}
+      return false
+      elsif arg.nil?
+        my_each.to_a {|idx| return true if idx}
+      elsif !arg.nil && (arg.is_a? class)
+        my_each.to_a {|idx| return true if [idx.class, idx.class.superclass].include?(arg)}
+      elsif !arg.nil && arg.class == regexp
+        my_each.to_a {|idx| return true if arg.match(idx)}
+      else
+        my_each.to_a {|idx| return true if idx == arg}
+    end
+    false
+  end
+
+  def my_none(arg = nil, &block)
+    !my_any(arg, &block)
+  end
+
+  def my_count(*arg)
+    too_manny_args_error?(arg)
+    result = 0
+    my_each do |idx|
+      if arg.size == 1
+        result += 1 if arg[0] == idx
+      elsif block_given?
+        result += 1 if yield(idx)
+      elsif arg.size.zero?
+        result += 1
+      end
+    end
+    result
+  end
```

Comment:
- [ ] `my_count` should return the number of items in enum through enumeration.
Your code throws the following error 👇 
![Screen Shot 2020-10-09 at 7 29 05 PM](https://user-images.githubusercontent.com/48804258/95613733-b2a40380-0a65-11eb-8e63-29aaeb98b551.png)
when given the following arguments 👇 
```ruby
array = [1,2,3,4,56,6,7,53,23,45,1]
range = Range.new(5,50)
p array.my_count
p range.my_count
p array.my_count(&block )
p array.my_count(1)
```